src/go/k8s/apis/redpanda/v1alpha1/cluster_types.go

```diff
@@ -618,6 +617,29 @@ func (p PandaproxyAPI) GetTLS() *TLSConfig {
 	}
 }
 
+// GetRedpandaResources returns resources that will be used by redpanda
+// container. This takes the total amount of resources given in the crd minus
+// all CRDs assigned to all enabled sidecars
+func (r *Cluster) GetRedpandaResources() corev1.ResourceRequirements {
+	assignedResources := *r.Spec.Resources.DeepCopy()
+	if r.Spec.Sidecars.RpkStatus != nil && r.Spec.Sidecars.RpkStatus.Enabled && r.Spec.Sidecars.RpkStatus.Resources != nil {
```

Comment:
Is it possible to have `RpkStatus` enabled but `RpkStatus.Resources` nil?

Reply:
I think if you're running without webhook it's possible. Better be safe than sorry 😛 because it's pointer and omitempty